docs/spec/concepts.rst

```diff
@@ -3,6 +3,272 @@
+:ref:`Any` to all expressions, and should emit no errors. Inferring static
+types or adding type annotations to the program (making the program more
+statically typed) may result in static type errors, if the program is not
+correct or if the static types aren't able to fully represent the runtime
+types. Removing type annotations (making the program more dynamic) should not
+result in additional static type errors. This is often referred to as the
+**gradual guarantee**.
+
+In Python's type system, we don't take the gradual guarantee as a strict
+requirement, but it's a useful guideline.
+
+Subtype, supertype, and type equivalence
+----------------------------------------
+
+A fully static type ``B`` is a **subtype** of another fully static type ``A``
+if and only if the set of values represented by ``B`` is a subset of the set of
+values represented by ``A``. Because the subset relation on sets is transitive
+and reflexive, the subtype relation is also transitive (if ``C`` is a subtype
+of ``B`` and ``B`` is a subtype of ``A``, then ``C`` is a subtype of ``A``) and
+reflexive (``A`` is always a subtype of ``A``).
+
+The **supertype** relation is the inverse of subtype: ``A`` is a supertype of
+``B`` if and only if ``B`` is a subtype of ``A``; or equivalently, if and only
+if the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on fully static types: the types
+``A`` and ``B`` are equivalent (or "the same type") if and only if ``A`` is a
+subtype of ``B`` and ``B`` is a subtype of ``A``. This means that the set of
+values represented by ``A`` is both a superset and a subset of the values
+represented by ``B``, meaning ``A`` and ``B`` must represent the same set of
+values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` (or as a "proper
+subtype" of ``A``) if ``B`` is a subtype of ``A`` and ``B`` is not equivalent
+to ``A``. In the same scenario we can describe the type ``A`` as "wider" than
+``B``, or a "proper supertype" of ``B``.
+
+Materialization
+---------------
+
+Since :ref:`Any` represents an unknown static type, it does not represent any
+known single set of values (it represents an unknown set of values). Thus it is
+not in the domain of the subtype, supertype, or equivalence relations on static
+types described above.
+
+To relate gradual types more generally, we define a **materialization**
+relation. Materialization transforms a "more dynamic" type to a "more static"
+type. Given a gradual type ``A``, if we replace zero or more occurrences of
+``Any`` in ``A`` with some gradual type (which can be different for each
+occurrence of ``Any``), the resulting gradual type ``B`` is a materialization
+of ``A``. (We can also materialize a :ref:`Callable` type by replacing ``...``
+with a bracketed list of types, and materialize ``tuple[Any, ...]`` by
```

Comment:
```suggestion
of ``A``. (We can also materialize a :ref:`Callable` type by replacing it
with a callable types with a fully static parameter list, and materialize ``tuple[Any, ...]`` by
```
Nit: Materializing `Callable[..., Any]` doesn't necessarily mean replacing the `...` with a "bracketed list of types", because such a bracketed list can only express positional-only parameters. For example, given `f: Callable[..., int]`, a call `f(x=3)` is valid because the type of `f` could be materialized to a callable type that accepts an `x` keyword parameter.